streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDbIterator.java

```diff
@@ -58,7 +61,11 @@ public synchronized boolean hasNext() {
             return allDone();
         } else {
             next = getKeyValue();
-            iter.next();
+            if (reverse) {
```

Comment:
Instead of storing and checking the `reverse` flag on every iteration, can we define something like
```
java.util.function.Consumer<RocksIterator> advanceIterator = reverse ? RocksIterator::prev : RocksIterator::next;
```
so then we can just blindly call `advanceIterator.accept(iter)`